application/src/main/java/org/togetherjava/tjbot/commands/UserInteractorPrefix.java

```diff
@@ -0,0 +1,60 @@
+package org.togetherjava.tjbot.commands;
+
+import org.jetbrains.annotations.Contract;
+
+/**
+ * Contains the prefixes for the UserInteractor's.
+ * <p>
+ * This is used for separate interactors with the same name, by command type (and possibly more in
+ * the future). Our system doesn't allow multiple interactors with the same name, while having a
+ * slash-command and a message-context-command with the same name can be really useful.
+ */
+public enum UserInteractorPrefix {
+    SLASH_COMMAND("s-"),
+    MESSAGE_CONTEXT_COMMAND("mc-"),
+    USER_CONTEXT_COMMAND("uc-");
+
+    private final String prefix;
+
+    UserInteractorPrefix(final String prefix) {
+        this.prefix = prefix;
+    }
+
+    public String getPrefix() {
+        return prefix;
+    }
+
+    /**
+     * Checks what enum value the given instance collaborates to.
+     * <p>
+     * As example,
+     *
+     * @param instance an instance to type check for a prefix
+     * @param <T> the type of the instance
+     * @return the prefixed {@link String}
+     */
+    public static <T extends UserInteractor> String getPrefixFromInstance(final T instance) {
```

Comment:
Maybe Optional might be easier to understand than ""

Reply:
yes. thanks

Reply:
yes